src/main/c/Jep/convert_p2j.c

```diff
@@ -745,28 +784,16 @@ static jbyteArray pybuffer_as_jbytearray(JNIEnv *env, PyObject *pybuffer)
                      view.format);
     }
 
-    PyBuffer_Release(&view);
+    JepBuffer_Release(&view);
     return jarr;
 }
 
-static jshortArray pybufferview_as_jshortarray(JNIEnv *env, Py_buffer view)
+static jshortArray pybufferview_as_jshortarray(JNIEnv *env, JepBuffer view)
 {
-    Py_ssize_t length = view.len / view.itemsize;
-    jshortArray jarr = (*env)->NewShortArray(env, (jsize) length);
-    if (process_java_exception(env)) {
-        return NULL;
-    }
-    if (PyBuffer_IsContiguous(&view, 'A')) {
-        (*env)->SetShortArrayRegion(env, jarr, 0, (jsize) length, (jshort*) view.buf);
-    } else {
-        void* buf = (*env)->GetShortArrayElements(env, jarr, NULL);
-        PyBuffer_ToContiguous(buf, &view, view.len, 'C');
-        (*env)->ReleaseShortArrayElements(env, jarr, buf, 0);
-    }
-    return jarr;
+    pybufferview_as_array_IMPL(jshort, Short)
```

Comment:
Actually, these macros are good because they clean up the code duplication.......